src/Buffers/DequeBuffer.cc

```diff
@@ -0,0 +1,70 @@
+#include "app-framework/Buffers/DequeBuffer.hh"
+
+template <class T>
+appframework::DequeBuffer<T>::DequeBuffer(){
+  Configure();
+}
+
+template <class T>
+void appframework::DequeBuffer<T>::Configure(){
+  
+  this->fAttributes.isBounded = true;
+  this->fAttributes.isSearchable = false;
+  
+  fDeque = std::deque<T>();
+
+  if(this->attributes().isBounded){
+    fCapacity = 1000;
+  }
+  else{
+    fCapacity = fDeque.max_size();
+  }
+
+  this->fPushTimeout = 1e6;
+  this->fPopTimeout  = 1e7;
+  fRetryTime   = 100;
+
+  fPushRetries = this->fPushTimeout/fRetryTime;
+  fPopRetries  = this->fPopTimout/fRetryTime;
+}
+
+template <class T>
+int appframework::DequeBuffer<T>::push(const T& obj){
+
+  int n_retries=0;
+  while(full()){
+    if(n_retries==fPushRetries){
```

Comment:
What is the rationale of handling the retries internally, rather than letting the module to decide how many retries are acceptable?

Reply:
Both push and pop should have a timeout parameter to allow the UserModule pushing/popping an opportunity to go and do something else, give up, or throw an execption

Reply:
In this case the retries are a consequence of the timeout configuration parameter defined in Buffer{Input,Output}. This is a specific implementation detail, different Buffer implementations may have their own ways of handling timeouts in push/pop.

It is of course up to UserModules to determine what to do when a push/pop times out.